src/modules/test/main.c

```diff
@@ -418,21 +431,53 @@ int RunTestStep(const TEST_STEP* test, const MANAGEMENT_MODULE* module)
                     LOG_ERROR("Failed to parse JSON payload: %s", payloadString);
                     result = EINVAL;
                 }
+                else if ((0 == strcmp(test->component, SECURITY_BASELINE)) && 
+                    (0 == strncmp(test->object, audit, strlen(audit))))
+                {
+                    asbAudit = true;
+                    
+                    for (i = 0; i < numSkippedAudits; i++)
+                    {
+                        if (0 == strcmp(test->object, skippedAudits[i]))
+                        {
+                            asbAudit = false;
+                            break;
+                        }
+                    }
+                }
             }
         }
 
-        if (test->payload != NULL)
+        if (test->payload || asbAudit)
         {
-            if (actualJsonValue != NULL)
+            if (asbAudit)
+            {
+                if (NULL == (reason = json_value_get_string(actualJsonValue)))
+                {
+                    LOG_ERROR("Assertion failed, json_value_get_string('%s') failed", json_serialize_to_string(actualJsonValue));
+                    result = -1;
+                }
+                else if (0 != strncmp(reason, SECURITY_AUDIT_PASS, strlen(SECURITY_AUDIT_PASS)))
+                {
+                    LOG_ERROR("Assertion failed, expected: '%s...', actual: '%s'", SECURITY_AUDIT_PASS, reason);
+                    result = -1;
+                }
+                else
+                {
+                    LOG_INFO("Assertion passed with reason: '%s'", reason + strlen(SECURITY_AUDIT_PASS));
+                }
+            }
+            else if (actualJsonValue != NULL)
             {
                 if (NULL == (expectedJsonValue = json_parse_string(test->payload)))
                 {
                     LOG_ERROR("Failed to parse expected JSON payload: %s", test->payload);
-                    result = EINVAL;
+                    result = -1;
```

Comment:
`EINVAL` seems like a better choice here? Why change?

Reply:
For consistency. This test uses -1 for test pass failures and in the end does not matter. If rest of this test code would use real errno errors, sure. As it is this looked like a sore thumb.

Reply:
Replaced all -1 returns in this function with EFAULT